File: 조준장/level2/행렬 테두리 회전하기.py
def solution(rows, columns, queries):
    answer = []
    graph = []
    cnt = 1
    for r in range(rows):
        temp_graph = []
        for c in range(columns):
            temp_graph.append(cnt)
            cnt += 1
        graph.append(temp_graph)
        
    for q in queries:
        sx, sy, ex, ey = q
        sx -= 1
        sy -= 1
        ex -= 1
        ey -= 1
        min_value = 1e9
        new_graph = [[0]*columns for _ in range(rows)]
        
        # 왼쪽
        for x in range(sx, ex):
            new_graph[x][sy] = graph[x+1][sy] 
            min_value = min(min_value, graph[x+1][sy])
        
        # 아래
        for y in range(sy, ey):
            new_graph[ex][y] = graph[ex][y+1] 
            min_value = min(min_value, graph[ex][y+1])

        # 오른쪽
        for x in range(sx, ex):
            new_graph[x+1][ey] = graph[x][ey] 
            min_value = min(min_value, graph[x][ey])
        
        # 위
        for y in range(sy, ey):
            new_graph[sx][y+1] = graph[sx][y] 
            min_value = min(min_value, graph[sx][y])

        for i in range(rows):
            for j in range(columns):
                if new_graph[i][j] != 0:
                    graph[i][j] = new_graph[i][j]

        answer.append(min_value)
    
    return answer  

File: 조준장/level2/행렬 테두리 회전하기.kt
class Solution {
    fun solution(rows: Int, columns: Int, queries: Array<IntArray>): IntArray {
        val answer = mutableListOf<Int>()
        
        val graph : MutableList<MutableList<Int>> = mutableListOf()
        
        var cnt = 1
        repeat(rows) { i ->
            val tempGraph = mutableListOf<Int>()
            repeat(columns) { j ->
                tempGraph.add(cnt)
                cnt++
            }
            graph.add(tempGraph)
        }
        
        for (querie in queries) {
            val sx = querie[0] - 1
            val sy = querie[1] - 1
            val ex = querie[2] - 1
            val ey = querie[3] - 1
            
            val newGraph = MutableList(rows) { MutableList(columns) { 0 } }
            var minValue = Int.MAX_VALUE
            
            // 오른쪽
            for (ny in sy until ey) {
                newGraph[sx][ny + 1] = graph[sx][ny]
                minValue = minOf(minValue, graph[sx][ny])
            }
            
            // 아래
            for (nx in sx until ex) {
                newGraph[nx + 1][ey] = graph[nx][ey]
                minValue = minOf(minValue, graph[nx][ey])
            }

            // 왼쪽
            for (ny in sy until ey) {
                newGraph[ex][ny] = graph[ex][ny + 1]
                minValue = minOf(minValue, graph[ex][ny + 1])
            }

            // 위로
            for (nx in sx until ex) {
                newGraph[nx][sy] = graph[nx + 1][sy]
                minValue = minOf(minValue, graph[nx + 1][sy])
            }

            repeat(rows) { i ->
                repeat(columns) { j ->
                    if(newGraph[i][j] != 0) {
                        graph[i][j] = newGraph[i][j]
                    }
                    
                }
            }
            answer.add(minValue)
        }
        
        return answer.toIntArray()
    }
}
